tests/test_optimizers.py

```diff
@@ -215,3 +227,213 @@ class TestFtrlOptimizer(GPflowTestCase, OptimizerCase):
+    Y = np.random.randn(N, 1)
+    kern = gpflow.kernels.RBF(D)
+    lik_var = 0.1
+    lik = gpflow.likelihoods.Gaussian()
+    lik.variance = lik_var
+
+    m_vgp = gpflow.models.VGP(X, Y, kern, lik)
+    m_gpr = gpflow.models.GPR(X, Y, kern)
+    m_gpr.likelihood.variance = lik_var
+
+    m_vgp.set_trainable(False)
+    m_vgp.q_mu.set_trainable(True)
+    m_vgp.q_sqrt.set_trainable(True)
+    NatGradOptimizer(1.).minimize(m_vgp, [[m_vgp.q_mu, m_vgp.q_sqrt]], maxiter=1)
+
+    assert_allclose(m_gpr.compute_log_likelihood(),
+                    m_vgp.compute_log_likelihood(), atol=1e-5)
+
+
+def test_other_XiTransform_VGP_vs_GPR(session_tf, xi_transform=XiSqrtMeanVar()):
+    """
+    With other transforms the solution is not given in a single step, but it should still give the same answer
+    after a number of smaller steps. 
+    """
+    N, D = 3, 2
+    X = np.random.randn(N, D)
+    Y = np.random.randn(N, 1)
+    kern = gpflow.kernels.RBF(D)
+    lik_var = 0.1
+    lik = gpflow.likelihoods.Gaussian()
+    lik.variance = lik_var
+
+    m_vgp = gpflow.models.VGP(X, Y, kern, lik)
+    m_gpr = gpflow.models.GPR(X, Y, kern)
+    m_gpr.likelihood.variance = lik_var
+
+    m_vgp.set_trainable(False)
+    m_vgp.q_mu.set_trainable(True)
+    m_vgp.q_sqrt.set_trainable(True)
+    NatGradOptimizer(0.01).minimize(m_vgp, [[m_vgp.q_mu, m_vgp.q_sqrt, xi_transform]], maxiter=500)
+
+    assert_allclose(m_gpr.compute_log_likelihood(),
+                    m_vgp.compute_log_likelihood(), atol=1e-4)
+
+
+def test_XiEtas_VGP_vs_GPR(session_tf):
+    class XiEta:
+        def meanvarsqrt_to_xi(self, mean, varsqrt):
+            return meanvarsqrt_to_expectation(mean, varsqrt)
+
+        def xi_to_meanvarsqrt(self, xi_1, xi_2):
+            return expectation_to_meanvarsqrt(xi_1, xi_2)
+
+        def naturals_to_xi(self, nat_1, nat_2):
+            return natural_to_expectation(nat_1, nat_2)
+
+    test_other_XiTransform_VGP_vs_GPR(session_tf, xi_transform=XiEta())
+
+
+def test_SVGP_vs_SGPR(session_tf):
+    """
+    With a Gaussian likelihood the sparse Gaussian variational (SVGP) model should be equivalent to the analytically 
+     optimial sparse regression model (SGPR) after a single nat grad step of size 1
+    """
+    N, M, D = 4, 3, 2
+    X = np.random.randn(N, D)
+    Z = np.random.randn(M, D)
+    Y = np.random.randn(N, 1)
+    kern = gpflow.kernels.RBF(D)
+    lik_var = 0.1
+    lik = gpflow.likelihoods.Gaussian()
+    lik.variance = lik_var
+
+    m_svgp = gpflow.models.SVGP(X, Y, kern, lik, Z=Z)
+    m_sgpr = gpflow.models.SGPR(X, Y, kern, Z=Z)
+    m_sgpr.likelihood.variance = lik_var
+
+    m_svgp.set_trainable(False)
+    m_svgp.q_mu.set_trainable(True)
+    m_svgp.q_sqrt.set_trainable(True)
+    NatGradOptimizer(1.).minimize(m_svgp, [[m_svgp.q_mu, m_svgp.q_sqrt]], maxiter=1)
+
+    assert_allclose(m_sgpr.compute_log_likelihood(),
+                    m_svgp.compute_log_likelihood(), atol=1e-5)
+
+
+class CombinationOptimizer(Optimizer):
```

Comment:
Shouldn't this class live in the `natgrad_optimizer` module?

Reply:
This is just for testing purposes and also serves as a demo for how the Actions might work. When Actions is done this won't be necessary

Reply:
@stefanosele, we will remove it as soon as Actions are merged into master.